acme/acme/client.py

```diff
@@ -519,7 +519,12 @@ def __init__(self, key, alg=jose.RS256, verify_ssl=True,
         self._default_timeout = timeout
 
     def __del__(self):
-        self.session.close()
+        # Try to close the session, but don't show exceptions to the
+        # user if the call to close() fails. See #4840.
+        try:
+            self.session.close()
+        except:  # pylint: disable=bare-except
```

Comment:
My thought was we want to make a best effort attempt to properly close the session. Using `__del__` is very error prone and we don't want to print tracebacks for any strange errors that might occur. By using the bare except, we're protected from dealing with this problem again if other errors manifest in the future.

Perhaps we want to know what other exceptions are possible though and want the bug reports. I don't really have a strong preference though and created #4894 with this change. Feel free to approve whichever one you like better.